test/integration/job/job_test.go

```diff
@@ -1520,16 +1531,18 @@ func TestDelayTerminalPhaseCondition(t *testing.T) {
 			featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.ElasticIndexedJob, true)
 			featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.JobSuccessPolicy, test.enableJobSuccessPolicy)
 
-			closeFn, restConfig, clientSet, ns := setup(t, "delay-terminal-condition")
-			t.Cleanup(closeFn)
 			ctx, cancel := startJobControllerAndWaitForCaches(t, restConfig)
 			t.Cleanup(cancel)
 
+			test.job.Name = fmt.Sprintf("test-job-%d", i)
 			jobObj, err := createJobWithDefaults(ctx, clientSet, ns.Name, &test.job)
 			if err != nil {
 				t.Fatalf("Error %q while creating the job %q", err, jobObj.Name)
 			}
-			t.Cleanup(func() { removePodsFinalizer(ctx, t, clientSet, ns.Name) })
+			t.Cleanup(func() {
+				removePodsFinalizerByName(ctx, t, clientSet, ns.Name, "fake.example.com/blockDeletion")
```

Comment:
@mimowo is it worth the extra effort to remove multiple finalizers in a single call?
Something like:
```
removePodsFinalizersByName(ctx, t, clientSet, ns.Name, []string{"fake.example.com/blockDeletion", "batch.kubernetes.io/job-tracking"})
```

Reply:
I don't have a strong opinion about this, it came out of a discussion with @mimowo on Slack. You could argue that having different sequential calls makes more explicit to the reader the order in which the finalizers should be removed, but it's a brittle argument...

Reply:
> @mimowo is it worth the extra effort to remove multiple finalizers in a single call? Something like:
> 
> ```
> removePodsFinalizersByName(ctx, t, clientSet, ns.Name, []string{"fake.example.com/blockDeletion", "batch.kubernetes.io/job-tracking"})
> ```

I like the proposal.

Reply:
> You could argue that having different sequential calls makes more explicit to the reader the order in which the finalizers should be removed 

I was only concerned with doing `Finalizers = nil`, as this could hide some potentially important details.

Reply:
ok! so I'll implement it 😊